### ISSUE TITLE
Errors with `TypeError: type 'typing.TypeVar' is not an acceptable base type`

### ISSUE THREAD

**Issue by gegoune** [tfranzel/drf-spectacular#1137]
**Describe the bug**
Dear maintainers, I tried searching closed and open issues as well as PRs and couldn't find anything related, hopefully it's not a duplicate issue.

I have just tried to install drf-spectacular after reading drf's documentation stating their built in openschema generator is deprecated.

My Django project is unable to start with drf-spectacular installed and configured as per readme. Traceback is

```
Performing system checks...

Exception in thread django-main-thread:
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/threading.py", line 1052, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.12/threading.py", line 989, in run
    self._target(*self._args, **self._kwargs)
  File "/venv/lib/python3.12/site-packages/django/utils/autoreload.py", line 64, in wrapper
    fn(*args, **kwargs)
  File "/venv/lib/python3.12/site-packages/django/core/management/commands/runserver.py", line 133, in inner_run
    self.check(display_num_errors=True)
  File "/venv/lib/python3.12/site-packages/django/core/management/base.py", line 485, in check
    all_issues = checks.run_checks(
                 ^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.12/site-packages/django/core/checks/registry.py", line 88, in run_checks
    new_errors = check(app_configs=app_configs, databases=databases)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.12/site-packages/django/core/checks/urls.py", line 14, in check_url_config
    return check_resolver(resolver)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.12/site-packages/django/core/checks/urls.py", line 24, in check_resolver
    return check_method()
           ^^^^^^^^^^^^^^
  File "/venv/lib/python3.12/site-packages/django/urls/resolvers.py", line 516, in check
    for pattern in self.url_patterns:
                   ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.12/site-packages/django/utils/functional.py", line 47, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
                                         ^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.12/site-packages/django/urls/resolvers.py", line 735, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
                       ^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.12/site-packages/django/utils/functional.py", line 47, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
                                         ^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.12/site-packages/django/urls/resolvers.py", line 728, in urlconf_module
    return import_module(self.urlconf_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1381, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1354, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1325, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 929, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 994, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/app/clog/urls.py", line 1, in <module>
    from drf_spectacular.views import (
  File "/venv/lib/python3.12/site-packages/drf_spectacular/views.py", line 17, in <module>
    from drf_spectacular.plumbing import get_relative_url, set_query_parameters
  File "/venv/lib/python3.12/site-packages/drf_spectacular/plumbing.py", line 74, in <module>
    from typing_extensions import Literal as _PxLiteral
  File "/venv/lib/python3.12/site-packages/typing_extensions.py", line 1174, in <module>
    class TypeVar(typing.TypeVar, _DefaultMixin, _root=True):
TypeError: type 'typing.TypeVar' is not an acceptable base type
```

**To Reproduce**
Please let me know if I should come up with full Django project replicating issue, I only tried on my single project so far.

1. Install `drf-spectacular==0.27.0` (I have also tried `0.26.5` with same result)
2. add `"drf_spectacular"` to `INSTALLED_APPS`
3. add `"DEFAULT_SCHEMA_CLASS": "drf_spectacular.openapi.AutoSchema"` to `REST_FRAMEWORK`
4. add url paths:
```
from drf_spectacular.views import SpectacularAPIView, SpectacularRedocView, SpectacularSwaggerView

urlpatterns = [
  …
    path("schema", SpectacularAPIView.as_view(), name="schema"),
    path(
        "schema/swagger-ui",
        SpectacularSwaggerView.as_view(url_name="schema"),
        name="swagger-ui",
    ),
    path("schema/redoc", SpectacularRedocView.as_view(url_name="schema"), name="redoc"),
]
```

I can confirm that it's step 3. that causes Django to crash. 

**Expected behavior**
No crash on startup and serving schemas with frontends as expected.

I have `Django 5.0` and `Python 3.12.0`, `drf 3.14.0`.

Please let me know if you would like any more information

**Comment by tfranzel** [comment]
Hi,

Hi! Looks like you are currently using a broken version of `typing_extensions` (~ 4.5.0): https://github.com/python/typing_extensions/issues/243, which was fixed some time ago

For python 3.12, this is not a requirement for us:

https://github.com/tfranzel/drf-spectacular/blob/c53b6779ba1a2a577a8650145518021af343797a/requirements/base.txt#L7

so you seem to pull it from somewhere else. Using a more recent version (e.g. 4.9.0) should resolve the issue.

**Comment by gegoune** [comment]
Thank you! It was `django-stubs-ext` and `django-stubs` pulling it in. All resolved with 4.9.0. Really appreciated!
